spark/src/main/scala/org/apache/spark/sql/delta/files/DeltaFileFormatWriter.scala

```diff
@@ -377,6 +378,13 @@ object DeltaFileFormatWriter extends LoggingShims {
     }
   }
 
+   class PartitionedTaskAttemptContextImpl(conf: Configuration,
```

Comment:
Needed a way to actually send the types of the partition columns as part of the task context, that's important to distinguish the timestamp vs timestamp_ntz cases for this change.

Reply:
If there's another way to get the partition column type details without this addition, that'd be awesome but didn't see any where else to get these detials